### ISSUE TITLE
Radxa 3W no wifi avaliable

### ISSUE THREAD

**Issue by JohnPAguirre** [OpenIPC/sbc-groundstations#8]
I ran into an issue setting up the image for the Radxa 3W sbc that would not recognize the internal wifi card and it was showing up as the unmanaged card rt10. This manifested itself by nmtui not showing any available networks and wifi-connect.sh not actually connecting. This screenshot was with only a keyboard connected of the output of nmcli
![image](https://github.com/user-attachments/assets/f6d788e2-f686-40db-9b7b-68665f4702ab)

I finally figured out a fix that the address seems to be different for my board. 

Needed to replace the contents of 99-custom-wifi.rules to point to a different address.
Following this instructions here:
[https://github.com/.../radxa_pi_zero_3w/headless_setup.md...](https://github.com/OpenIPC/sbc-groundstations/blob/master/radxa_pi_zero_3w/headless_setup.md?fbclid=IwZXh0bgNhZW0CMTAAAR1he2B2w8w2-nCWQ7wxNFo5gVL1VxXp11cF8KMs_NkWL2PKixgIobdaC6Q_aem_SxvGh-HSYuqCJuFpWsiANA#setup-of-autoconnect-on-boot)

sudo nano /etc/udev/rules.d/99-custom-wifi.rules

SUBSYSTEM=="net", ACTION=="add", ATTR{address}=="38:7A:cc:*:*:*", NAME="wlan1"

I hope this helps the next person if their internal wifi card is not setup 

How I figured it out for the next person - nmcli to grab the device address and added it to the 99 custom rules file. Reboot and started getting bluetooth messages in dmesg. nmcli had another card as rt10 with a different address. Changed the address to this one, rebooted and now nmcli / nmtui allowed me to view and connect to a wireless network

**Comment by seriyps** [comment]
Yup, I had the same problem. Unfortunately, only decided to search in github only after I figured out the fix myself.

I think for the future fix, just both MAC addresses can be added to the rule file to support both older and newer boards.

**Comment by notsudogood** [comment]
hi i ran into the same problem , maybe because im new to the whole SBC stuff i dont think i implemented the fix correctly , I have attached an image of what i done ,once i wrote it i done ctrl+X to exit and then the Y key and then the ENTER key after which i done sudo reboot but i still could not get anything to show when i done nmtui or nmcli , it shows that rt10 is still unmanaged - not really sure where to go from here
![DSC_0238](https://github.com/user-attachments/assets/ac9467c9-881c-45bc-9cd2-934de10373b2)

**Comment by ramkrishna757575** [comment]
Here is a script that can be setup on the SBC to automatically detect the internal wifi adapter. This will make sure your wifi connection always works. It makes also possible to just hot-plug the wfb-ng adapters(multiple adapters can be plugged-in or removed). No need to change any config files and no need to reboot.

https://github.com/ramkrishna757575/hot-plug-wfb-nics/blob/master/README.md

In case you are unable to connect your SBC to your wifi network, you can put the sd-card in a card reader, mount the `rootfs` partition in your pc, and you can copy the script files in that.

**Comment by JohnPAguirre** [comment]
Looks like an amazing solution - Will test it on my local Radxa and I am hoping this gets baked into future image releases soon. Thanks so much!

**Comment by Smurfgit** [comment]
I must be missing something 
No luck with cli commands

**Comment by JohnDGodwin** [comment]
hotplugging is now implemented.
